packages/ethers/src/client.ts

```diff
@@ -273,6 +278,67 @@ export class Web3Modal extends Web3ModalScaffold {
         })
 
         return signature as `0x${string}`
+      },
+
+      parseUnits: (value: string, decimals: number) => parseUnits(value, decimals),
+
+      formatUnits: (value: bigint, decimals: number) => formatUnits(value, decimals),
+
+      async getEstimatedGas(data) {
+        const chainId = EthersStoreUtil.state.chainId
+        const provider = EthersStoreUtil.state.provider
+        const address = EthersStoreUtil.state.address
+
+        if (!provider) {
+          throw new Error('connectionControllerClient:sendTransaction - provider is undefined')
```

Comment:
Should we have custom errors?

Reply:
Wdym by 'custom errors'? Having better error messages? @lukaisailovic

Reply:
Custom error as in extending the error class eg.
```
class ConnectionControllerError extends Error {
...
}
```

I think it'd be a nice addition to standardize errors but I'd do it globally  (for all errors in w3m)

Reply:
@tomiir would be a nice addition but since we are following the same pattern for all error messages in the clients, let's set this as a error management improvement task. Created: https://linear.app/walletconnect/issue/W3M-257/improve-error-management-in-the-w3m-clients